content/client-server-api/modules/threading.md

```diff
@@ -0,0 +1,201 @@
+---
+type: module
+---
+
+### Threading
+
+{{% added-in v="1.4" %}}
+
+Threads allow users to visually branch their conversations in a room. Typically most used
+when a room is discussing multiple topics, threads provide more organisation of communication
+that traditional [rich replies](#rich-replies) can't always offer.
+
+Clients SHOULD render threads differently to regular messages or replies in the timeline, such
+as by providing some context to what is going on in the thread but keeping the full conversation
+history behind a disclosure.
+
+Threads are established using a `rel_type` of `m.thread` and reference the *thread root* (the
+first event in a thread). It is not possible to create a thread from an event with a `rel_type`,
+which includes not being able to nest threads. All conversation in a thread reference the thread
+root instead of the most recent message, unlike rich reply chains.
+
+As a worked example, the following represents a thread and how it'd be formed:
+
+```json
```

Comment:
Should we use `json5` or `jsonc` to highlight syntax here and in other code examples?